### PR TITLE
Add script to compare target version with latest Docker Hub tag.

### PR DESCRIPTION
**Pull request by jefchien**
**Description:** Replaced workflow dispatch inputs with script. The script pulls all the tags for the repo from Docker Hub and gets the latest version from there. Compares that to the target version and emits github workflow step variables that are used to determine whether to tag the new release as latest or mirror the operator.

**Link to tracking Issue:** <Issue number if applicable>

**Testing:** Ran the script locally.

**Documentation:** <Describe the documentation added.>

### CODE REVIEW

File: .github/workflows/CD.yml [hunk 1]
```diff
@@ -376,12 +372,11 @@ jobs:
           REPO_NAME="$IMAGE_NAMESPACE/$IMAGE_NAME"
           docker tag $IMAGE_NAME $REPO_NAME:$TAG
           docker push $REPO_NAME:$TAG
-          if [ ${{ github.event.inputs.update-latest }} == 'true' ]; then
+          if [ ${{ steps.version.outputs.any-update }} == 'true' ]; then
```
[the remark this answers is not in the document]
Comment (jefchien) [reply .github/workflows/CD.yml:375]: A new patch release for the current version would be the latest as well. We've made the decision to only mirror the operator on minor releases at minimum.

File: tools/release/s3-release.sh [hunk 2]
```diff
@@ -11,9 +13,6 @@
 # express or implied. See the License for the specific language governing
 # permissions and limitations under the License.
 
-
```
Comment (sethAmazon) [inline tools/release/s3-release.sh:14]: Good catch shebang should always be first line

File: tools/workflow/docker-version-compare.sh [hunk 3]
```diff
@@ -0,0 +1,95 @@
+#! /bin/bash
```
Comment (sethAmazon) [inline tools/workflow/docker-version-compare.sh:1]: Should we write this in go?

**Comment by sethAmazon** [comment]
Can you create some keys and test this on you private fork?

File: tools/workflow/docker-version-compare.sh (hunk 3 above)
Comment (jefchien) [reply tools/workflow/docker-version-compare.sh:1]: Thought about it, but it seemed simpler this way.

**Comment by jefchien** [comment]
Ran it on my fork by reducing the PR build workflow to just run the script I'm adding.
```sh
jobs:
  test:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v2
      - name: Compare version with Dockerhub latest
        id: version
        run: |
          TAG=v0.16.0
          TARGET_VERSION=$TAG bash tools/workflow/docker-version-compare.sh
      - name: Print out flags
        run: |
          echo "Any: ${{ steps.version.outputs.any-update }}"
          echo "Major: ${{ steps.version.outputs.major-update }}"
          echo "Minor: ${{ steps.version.outputs.minor-update }}"
          echo "Patch: ${{ steps.version.outputs.patch-update }}"
      - name: Test flags
        run: |
          if [ ${{ steps.version.outputs.any-update }} == 'true' ]; then
            echo "Update latest"
          fi
          if [ ${{ steps.version.outputs.major-update }} == 'true' ] || [ ${{ steps.version.outputs.minor-update }} == 'true' ]; then
            echo "Mirror operator"
          fi
```

image

**Pull request merged**